tests/dragonfly/cluster_test.py

```diff
@@ -1504,6 +1517,77 @@ async def test_cluster_config_reapply(df_factory: DflyInstanceFactory):
     await close_clients(*[node.client for node in nodes], *[node.admin_client for node in nodes])
 
 
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_replication_migration(
+    df_factory: DflyInstanceFactory, df_seeder_factory: DflySeederFactory
+):
+    """
+    Test replication with migration. Create the following setup:
+
+    master_1 -> replica_1, master_2 -> replica_2
+
+    with each master owning half the slots. Let them then fully exchange their slots
+    and make sure the captures on the replicas are equal.
+    """
+    instances = [
+        df_factory.create(port=BASE_PORT + i, admin_port=BASE_PORT + 1000 + i) for i in range(4)
+    ]
+    df_factory.start_all(instances)
+    m1, r1, m2, r2 = instances
+
+    nodes = [await create_node_info(n) for n in instances]
+    m1_node, r1_node, m2_node, r2_node = nodes
+    master_nodes = [m1_node, m2_node]
+
+    # divide node slots by half
+    m1_node.slots = [(0, 8000)]
+    m1_node.replicas = [r1_node]
+    m2_node.slots = [(8001, 16383)]
+    m2_node.replicas = [r2_node]
+
+    # push this config
+    await push_config(
+        json.dumps(generate_config(master_nodes)), [node.admin_client for node in nodes]
+    )
```

Comment:
Why do you push only to masters?
Our definition is that all nodes must have the same config, including replicas

Reply:
I push to all nodes, but generate the config only from masters perspective